test/externalTests/exttest/common.py

```diff
@@ -0,0 +1,357 @@
+        raise FileNotFoundError("package.json not found.")
+    package_json = package_json_path.read_text(encoding="utf-8")
+    package_json = re.sub(
+        r'("prepublish":)\s".+"',
+        lambda m: f'{m.group(1)} ""',
+        package_json
+    )
+    package_json = re.sub(
+        r'("prepare":)\s".+"',
+        lambda m: f'{m.group(1)} ""',
+        package_json
+    )
+    with open(package_json_path, "w", encoding="utf-8") as f:
+        f.write(package_json)
+
+
+def replace_version_pragmas(test_dir: Path):
+    """
+    Replace fixed-version pragmas (part of Consensys best practice).
+    Include all directories to also cover node dependencies.
+    """
+    print("Replacing fixed-version pragmas...")
+    for source in test_dir.glob("**/*.sol"):
+        content = source.read_text(encoding="utf-8")
+        content = re.sub(r"pragma solidity [^;]+;", r"pragma solidity >=0.0;", content)
+        with open(source, "w", encoding="utf-8") as f:
+            f.write(content)
+
+
+def run_test(name: str, runner: TestRunner):
+    rconfig = runner.config
+    if rconfig.solc.binary_type not in ("native", "solcjs"):
+        raise InvalidConfigError(
+            f"Invalid solidity compiler binary type: {rconfig.solc.binary_type}"
+        )
+    if rconfig.solc.binary_type != "solcjs" and rconfig.solc.solcjs_src_dir != "":
+        raise InvalidConfigError(
+            f"""Invalid test configuration: 'native' mode cannot be used with 'solcjs_src_dir'.
+            Please use 'binary_type: solcjs' or unset: 'solcjs_src_dir: {rconfig.solc.solcjs_src_dir}'"""
+        )
+
+    print(f"Testing {name}...\n===========================")
+    with TemporaryDirectory(prefix=f"ext-test-{name}-") as tmp_dir:
+        test_dir = Path(tmp_dir) / "ext"
+        presets = rconfig.selected_presets()
+        print(f"Selected settings presets: {' '.join(map(str, presets))}")
+
+        # Configure solc compiler
+        solc_version = setup_solc(rconfig, test_dir)
+        print(f"Using compiler version {solc_version}")
+
+        # Download project
+        download_project(test_dir, rconfig.repo_url, rconfig.ref_type, rconfig.ref)
+
+        # Configure run environment
+        if rconfig.build_dependency == "nodejs":
+            prepare_node_env(test_dir)
+        runner.setup_environment(test_dir)
+
+        replace_version_pragmas(test_dir)
+
+        # Configure TestRunner instance
+        runner.compiler_settings(solc_version, presets)
+        for preset in rconfig.selected_presets():
+            print("Running compile function...")
+            runner.compile(solc_version, preset)
+            if (
+                os.environ.get("COMPILE_ONLY") == "1"
+                or preset in rconfig.compile_only_presets
+            ):
+                print("Skipping test function...")
+            else:
+                print("Running test function...")
+                runner.run_test(preset)
+            # TODO: store_benchmark_report
```

Comment:
```suggestion
            # TODO: store_benchmark_report # pylint: disable=fixme
```
In order to turn off the `pylint` warning.